### DISCUSSION TITLE
Disabling develocity?

### DISCUSSION THREAD

**Discussion by Bananeweizen** [openrewrite/rewrite#4262] [Q&A] [answered]
Over time, the error messages of the gradle develocity plugin have become more prominent. What's a simple way to completely disable that plugin as an external contributor? Even if it were possible to upload build scans and build results from my local machine to your develocity instance, I don't want that to happen on my company laptop.

![grafik](https://github.com/openrewrite/rewrite/assets/406876/8a263df2-2e45-4cb8-bf86-30bc4f8c37c6)

Bonus points for making that part of the general gradle configuration in all the repositories (but I have no idea if you can "recognize" which authors or machines belong to the core team in an automated fashion).

**Comment by timtebeek** [comment]
Hi @Bananeweizen ! Sorry about this; wasn't aware this is affected OSS contributors, but thanks for calling it out!

I'm not quite sure what the options are, but open to explore those with you. For a start we seem not as standardized in our use of develocity as I'd hoped; Do you recall which project you saw this on?

As you'll know the folks at Gradle have rebranded from "Gradle Enterprise" to "Develocity", and we're in the process of adopting those changes, in part with recipes. Here you see an example of a project that's already made the switch recently: https://github.com/openrewrite/rewrite-spring/commit/8ca31b6c54908b4b14c68a6c4e678952d2c4ce08

Other projects, like our rewrite-build-gradle-plugin still seem to use the older format: https://github.com/openrewrite/rewrite-build-gradle-plugin/blob/c9eddb0dec82d426ce24b6d39a764855aeff1465/settings.gradle.kts#L10

It might make sense to standardize our usage of develocity, and as part of that potentially pull that into a shared configuration that only pushes when an access key is defined. @shanman190 might have some insights and experience to share on this front, which we can then roll out.

**Reply by shanman190** [reply, marked as answer]
The previous method invocations to only publish when authenticated were removed during the migration from `gradleEnterprise` to `develocity` and just need to be brought back such that it only publishes when authenticated like had been configured before. 

Something like this should work in the `rewrite` repo:
```
develocity {
    buildScan {
        publishing {
            onlyIf {
                authenticated
            }
        }
    }
}
```

**Reply by timtebeek** [reply]
Thanks for the quick reply! Picked up for openrewrite/rewrite just now in https://github.com/openrewrite/rewrite/commit/637b1d16f2b919ecb418aa957e51de6b70bc0bd0

Likely needs to be added to other projects still; I'm not sure if we can configure that once in `rewrite-build-gradle-plugin` or if that then needs a separate settings plugin there first. I recall something along those lines.

**Reply by timtebeek** [reply]
Ok, search and replaced across some 25 projects already;
https://app.moderne.io/commits/b3a23027-46a1-471d-aa99-46793334d116

The others appear to still be using the outdated Gradle Enterprise, which we should migrate first.

**Reply by timtebeek** [reply]
I'll mark this as resolved for now; the remaining gradle enterprise to develocity migrations will use the patterns seen in the already migrated projects, and since you mentioned the errors only became more prominent in more recent versions I think it'll be less of an issue on those older projects.

Thanks again for reporting the issue, and thanks as always to Shannon for providing quick guidance on the fix.

**Reply by Bananeweizen** [reply]
Thanks a lot, local builds look readable again. :)
